src/manifest.c

```diff
@@ -530,9 +530,12 @@ static int retrieve_manifests(int current, int version, char *component, struct
 		goto out;
 	}
 
-	if (!signature_download_and_verify(url, filename)) {
-		unlink(filename);
-		goto out;
+	if (signing_enabled && !signature_download_and_verify(url, filename)) {
```

Comment:
Also, "url" here ends with .tar, but we're not going to be signing the .tar, only the Manifest.MoM.